03-python2-functions-instructors.ipynb

```diff
@@ -0,0 +1,923 @@
+    "<h3><span class=\"fa fa-certificate\"></span> Learning Objectives: </h3>\n",
+    "</div>\n",
+    "\n",
+    "- Define a function that takes parameters.\n",
+    "- Return a value from a function.\n",
+    "- Test and debug a function.\n",
+    "- Set default values for function parameters.\n",
+    "- Explain why we should divide programs into small, single-purpose functions."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "At this point, we’ve written code to draw some interesting features in our inflammation data, loop over all our data files to quickly draw these plots for each of them, and have Python make decisions based on what it sees in our data. But, our code is getting pretty long and complicated; what if we had thousands of datasets, and didn’t want to generate a figure for every single one? Commenting out the figure-drawing code is a nuisance. Also, what if we want to use that code again, on a different dataset or at a different point in our program? Cutting and pasting it is going to make our code get very long and very repetative, very quickly. We’d like a way to package our code so that it is easier to reuse, and Python provides for this by letting us define things called ‘functions’ - a shorthand way of re-executing longer pieces of code.\n",
+    "\n",
+    "Let’s start by defining a function fahr_to_kelvin that converts temperatures from Fahrenheit to Kelvin:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "# Let's get our import statements out of the way first\n",
+    "from __future__ import division, print_function\n",
+    "import numpy as np\n",
+    "import glob\n",
+    "import matplotlib.pyplot as plt\n",
+    "%matplotlib inline\n",
+    "\n",
+    "def fahr_to_kelvin(temp):\n",
+    "    return ((temp - 32) * (5/9)) + 273.15"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The function definition opens with the word `def`, which is followed by the name of the function and a parenthesized list of parameter names. The body of the function — the statements that are executed when it runs — is indented below the definition line, typically by four spaces.\n",
+    "\n",
+    "When we call the function, the values we pass to it are assigned to those variables so that we can use them inside the function. Inside the function, we use a return statement to send a result back to whoever asked for it.\n",
+    "\n",
+    "Let’s try running our function. Calling our own function is no different from calling any other function:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "print('freezing point of water:', fahr_to_kelvin(32))\n",
+    "print('boiling point of water:', fahr_to_kelvin(212))"
```

Comment:
I think these two examples would make more sense to me if they are the opposite... I know these temperatures in ºC (or K) but not in ºF, so I would like to convert them from K to F...